retsinformation.js

```diff
@@ -0,0 +1,182 @@
+{
+	"translatorID": "feef66bf-4b52-498f-a586-8e9a99dc07a0",
+	"label": "Retsinformation",
+	"creator": "Roald Frøsig",
+	"target": "^https?://(?:www\\.)?retsinformation\\.dk/",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "g",
+	"lastUpdated": "2016-01-27 14:38:37"
+}
+
+function detectWeb(doc, url){
+	if (ZU.xpathText(doc, '//div[@class="wrapper2"]/table[@id="ctl00_MainContent_ResultGrid1"]')) {
+		return "multiple";
+	} else if (url.indexOf("R0710") != -1) {
+		threeLetters = getKortNavn(doc, url)[0].substr(0,3);
+		if (/(DOM|AFG|KEN|UDT)/.test(threeLetters)) {
+			return "case";
+		};
+		if (/(EDP|ISP|FOU)/.test(threeLetters)) {
+			return "report";
+		};
+		if (/\d{3}/.test(threeLetters)) {
+			return "bill";
+		} else return "statute";
+	} else return false;
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		var items = new Object();
+		Z.debug("L26"+url);
+		Z.debug("L27"+url.match("R0700.aspx\\?r"));
+		if (url.match(/(R0900|R0920|R0930)/)) {
+			var titles = ZU.xpath(doc, '//table[@id="ctl00_MainContent_ResultGrid1"]/tbody/tr[@class!="th"]/td[1]/a[1]');
+		} else if (url.match(/(R0210|R0310)/)) {
+			var titles = ZU.xpath(doc, '//table[@id="ctl00_MainContent_ResultGrid1"]/tbody/tr[@class!="th"]/td[2]/a[1]');
+		} else if (url.match("R0700.aspx\\?res")) {
+			Z.debug("L132"+url);
+			var titles = ZU.xpath(doc, '//table[@id="ctl00_MainContent_ResultGrid1"]/tbody/tr[@class!="th"]/td[3]/a[1]');
+		} else if (url.match(/(R0220|R0415|R0700)/)) {
+			var titles = ZU.xpath(doc, '//table[@id="ctl00_MainContent_ResultGrid1"]/tbody/tr[@class!="th"]/td[4]/a[1]');
+		}
+		for (i = 0; i < titles.length; i++) {
+			items[titles[i].href] = clean(titles[i].text);
+		}
+		Z.selectItems(items, function(selectedItems) {
+			if (!selectedItems) return true;
+			var urls = new Array();
+			for (var i in selectedItems) {
+				urls.push(i);
+			}
+			ZU.processDocuments(urls, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	};
+}
+
+function clean(text) {
+	text = text.replace(/\n/g," ");
+	text = text.replace(/ +/g," ");
+	text = text.trim();
+	return text;
+}
+
+function scrape(doc, url) {
+	var type = detectWeb(doc, url);
```

Comment:
Instead of calling `detectWeb`, move the relevant logic out of `detectWeb` into something like `getItemType` and call it from here and `detectWeb`. `detectWeb` can return "multiple", which would not be right here.